### PR TITLE
Refactor drawer

### PR DESCRIPTION
**Pull request by elParaguayo**
An attempt to tidy up the Drawer objects.

@m-col this has not been rebased onto the wlroots 0.16 branch.

Draft for now so we can test etc.

Main change is to stop `clear` from writing to `XCBSurface` in x11 or `ImageSurface` in Wayland. Instead, everything is rendered to a single `RecordingSurface` and the backend then performs specific actions to render it to the window.

### CODE REVIEW

**Comment by elParaguayo** [comment]
@m-col @jwijenbergh - I've done a very ugly job of rebasing this onto #3985. Could you test this in conjunction with this branch of qtile-extras (https://github.com/elParaguayo/qtile-extras/tree/refactor-drawer-decorations) and see if decorations work ok for you.

I'll do some more testing here.

File: tox.ini [hunk 1]
```diff
@@ -103,20 +104,25 @@ deps =
...
     bowler
     dbus-next
-    xcffib >= 0.10.1
     PyGObject
     pytest >= 6.2.1
     types-python-dateutil
     types-pytz
     types-pkg_resources
 commands =
+<<<<<<< HEAD
```
Comment (jwijenbergh) [inline tox.ini:113]: leftovers from rebase

File: tox.ini [hunk 2]
```diff
@@ -103,20 +104,25 @@ deps =
...
     pip install -r requirements.txt pywayland>=0.4.14 xkbcommon>=0.3
+    pip install pywlroots>=0.16.1,<0.17.0
+=======
```
Comment (jwijenbergh) [inline tox.ini:116]: leftovers from rebase

File: tox.ini [hunk 3]
```diff
@@ -103,20 +104,25 @@ deps =
...
+    pip install --ignore-installed xcffib wheel
+    pip install --ignore-installed --no-cache --upgrade --no-build-isolation cairocffi
+    pip install pywayland>=0.4.14 xkbcommon>=0.3
     pip install pywlroots>=0.15.24,<0.16.0
+>>>>>>> 2b68c3da (cairocffi 1.5.0 build changes)
```
Comment (jwijenbergh) [inline tox.ini:121]: leftovers from rebase

**Pull request closed without merging**